src/energenieApi/energenieApi.ts

```diff
@@ -50,17 +51,20 @@ export class EnergenieApi {
       'getSubDevices');
   }
 
-  public toggleSocketPower(id: number, action: string): Promise<void> {
+  public toggleSocketPower(id: number, value: boolean): Promise<void> {
     const params = `params={ "id": ${id} }`;
+    const action = value ? 'power_on' : 'power_off';
+    const url = getUrl('subdevices' + action, this.baseUrl);
```

Comment:
Think this is missing a slash, should be `` `subdevices/${action}` ``

Reply:
Fixed.